app/neo/editors/form/useActionHandler.ts

```diff
@@ -20,15 +21,25 @@ const editorPath = (action: FormActionArgs, formEditorPath: string) => {
 export const useActionHandler = (project: ProjectIdentifier, formEditorPath: string) => {
   const { openEditor } = useEditors();
   const { createEditorFromPath } = useCreateEditor();
+  const { getComponentProject } = useGetComponentProject();
   const openUrl = useOpenUrl();
   return useCallback<FormActionHandler>(
     action => {
       if (action.actionId === 'openUrl') {
         openUrl(action.payload);
         return;
       }
+      if (action.actionId === 'openComponent') {
+        const parts = action.payload.split('.');
+        const transformedPayload = parts.join('/') + '/' + parts[parts.length - 1];
+        const editorPath = 'src_hd/' + `${transformedPayload}${FORM_EDITOR_SUFFIX}`;
+        getComponentProject({ project, id: action.payload }).then(p => {
```

Comment:
if this method would return a HdBean, you can simply create an editor of it and open it.